### DISCUSSION TITLE
x-show not working in some cases

### DISCUSSION THREAD

**Discussion by simonferndriger** [alpinejs/alpine#3955] [7. Bugs (archived)]
Unfortunately, I did not figure out (yet) the reason x-show is not working in a specific case, which is why I use `<template x-if>` instead. The part in the (production) code looks like this (with x-show):

`<section class="player sticky" id="webcam" @mouseenter="clearTimeout(timeouts.ad)" @mouseleave="if(!ad.show) timeouts.ad = true" x-bind="$" x-data="player('webcam')">`

    //--------------------------------------------------------------
    Alpine.data('player', id => ({
    //--------------------------------------------------------------
        // - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
        $: {
        // - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
            [':animate'](){
            //- - - - - - - - - - - - - - - - - -
                switch(id){
                    //---
                    case 'ad':
                    //---
                        if(this.ad.show === 'instant')
                            return '';
                        
                        return this.columns === 2 ? 'fadeInSlow' : 'fadeIn';
    
                    //---
                    case 'webcam':
                    //---
                        break;
                }
            },
            //- - - - - - - - - - - - - - - - - - 
            [':class'](){
            //- - - - - - - - - - - - - - - - - - 
                return {
                    //fixed: !this.visible.intro || this.view === 'map',
                    'hide-scrollbar': id === 'ad' || (this.view === 'raster' && this.columns === 2),
                    inactive: !this.ad.paused && !this.ad.show && this.columns === 2 && this.hovered.webcams && this.timeouts.ad,
                    'show-scrollbar': id === 'webcam' && (this.view === 'map' || this.columns === 1),
                    'overflow-hidden': this.visible.intro,
                };
            },
            //- - - - - - - - - - - - - - - - - - 
            ['x-show'](){
            //- - - - - - - - - - - - - - - - - - 
                if(id === 'webcam')
                    return this.webcam && !this.ad.show;
            },
        },

**Comment by simonferndriger** [comment]
There is no js error in the console. In total I use two instances of 'player'. In the first attempt, I had the `x-show`directive directly in the html tag, that didn't work either.

**Comment by simonferndriger** [comment]
https://github.com/alpinejs/alpine/discussions/3780:
This might explain why `pois` get rendered (which are within the `player('webcam')` instance) when using `<template x-if>` to hide it. If the condition of `x-show` is still true, then it makes sense that they still (are trying to) get rendered.

**Comment by ekwoka** [comment]
What is with these comments?

can you try having the x-show bind only return a string of `webcam && !ad.show` instead ? to test.

**Reply by simonferndriger** [reply]
_These comments (the dashed lines) just help me get a quick overview, sorry if that was confusing._

I tried leaving out `if(id === 'webcam')` and it does not change anything. The funny thing is, it works in case `webcam` was initially undefined and only gets a value after a user interaction. The case it does not work, is when webcam is assigned a value right from the start (i.e. landing on `/webcam/[id]`).

**Reply by ekwoka** [reply]
So did you try having it only return a string of the condition instead of a boolean?

**Reply by simonferndriger** [reply]
Ah sorry, I misunderstood your question. Yes, now I tried it like this and there was still the same issue. The data displayed for the webcam (`webcam?.prop`) is no more visible, but the container still is after setting `webcam = null`.

            if(id === 'webcam')
                return 'webcam && ad.show';
                
            return true;
